### PR TITLE
User

### PR DESCRIPTION
**Pull request by yjwangit**
Auth0 login setup

### CODE REVIEW

**Review by LeiaQuesada — APPROVED** [review]

**Review by m3ia — APPROVED** [review]
The rendering looks great so far! Nice job with separating the components in the components folder.

**Comment by LeiaQuesada** [comment]
Yes, agreed with Meia great job on integrating Auth0 and refactoring your functions!

**Comment by yjwangit** [comment]
Thank you Meia and Leia for your review!

File: app/package.json [hunk 1]
```diff
@@ -29,9 +29,12 @@
   },
   "proxy": "http://localhost:4000",
   "dependencies": {
+    "@auth0/auth0-react": "^1.4.0",
     "axios": "^0.21.1",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
+    "react-router": "^5.2.0",
+    "react-router-dom": "^5.2.0",
```
Comment (alexcheva) [inline app/package.json:37]: Nice!

File: app/src/index.js [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
+// const auth0CallbackUrl = "http://localhost:3000/"; // window.location.origin
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <Auth0Provider
+      domain="dev-m1so.us.auth0.com"
+      clientId="BTb7r8IKJVu9BnKFll6N19bjBfa8heqw"
+      redirectUri={window.location.origin}
+    >
```
Comment (alexcheva) [inline app/src/index.js:21]: we probably will need to put those in .env file later, as suggested in the blog post 😬

File: app/src/components/Results.js [hunk 3]
```diff
@@ -1,13 +1,38 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
```
Comment (alexcheva) [inline app/src/components/Results.js:1]: I would suggest using - import * as React from "react"; instead

**Review by alexcheva — APPROVED** [review]
Looks good! Great job!

File: app/src/components/Results.js (hunk 3 above)
Comment (yjwangit) [reply app/src/components/Results.js:1]: Thank you very much for your review and suggestions, Alex!

**Pull request merged**